From 7f8e7070e66ad2191e3d8f60e99e4b4bb40fb9f1 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Fri, 17 Feb 2023 10:34:08 +0800
Subject: [PATCH] u-boot-socfpga: fix boot from script boot.scr failed

The ostree image requires to run script boot.scr in u-boot,
After upgrading u-boot to v2021.07, the FIT formatting image is used
by default. it failed:

    => source
    ## Executing script at 80000000
    Wrong image format for "source" command

From doc/uImage.FIT/signature.txt:
WARNING: When relying on signed FIT images with required signature check
the legacy image format is default disabled by not defining
CONFIG_LEGACY_IMAGE_FORMAT

Set CONFIG_LEGACY_IMAGE_FORMAT=y to run script boot.scr in u-boot

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 configs/socfpga_stratix10_mmc_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configs/socfpga_stratix10_mmc_defconfig b/configs/socfpga_stratix10_mmc_defconfig
index 59fad46669..2bca280b3c 100644
--- a/configs/socfpga_stratix10_mmc_defconfig
+++ b/configs/socfpga_stratix10_mmc_defconfig
@@ -93,3 +93,4 @@ CONFIG_FAT_WRITE=y
 CONFIG_MENU=y
 CONFIG_CMD_BOOTMENU=y
 CONFIG_SYS_LOAD_ADDR=0x02000000
+CONFIG_LEGACY_IMAGE_FORMAT=y
-- 
2.25.1

